be/src/fs/fs_s3.cpp

```diff
@@ -106,11 +108,14 @@ class S3ClientFactory {
 
     class ClientCacheKey {
     public:
-        ClientConfiguration config;
-        AWSCloudConfiguration aws_cloud_configuration;
+        ClientConfigurationPtr config;
+        AWSCloudConfigurationPtr aws_cloud_configuration;
```

Comment:
I'm wondering if std::unique_ptr<> is suitable for it or not. If this config is not shared by many owners, std::unique_ptr is preferred.

Reply:
it is not suitable to use std::unique_ptr<>, because it is difficult to copy while ClientCacheKey will be assigned frequently and it is difficult to make ClientCacheKey copyable.